src/java/org/apache/cassandra/db/SinglePartitionReadCommand.java

```diff
@@ -957,6 +958,23 @@ private ClusteringIndexNamesFilter reduceFilter(ClusteringIndexNamesFilter filte
         }
 
         NavigableSet<Clustering<?>> toRemove = null;
+
+        DeletionInfo deletionInfo = result.deletionInfo();
```

Comment:
nit: I think all actual implementations of `Partition` would support it, so you could just add `result.deletionInfo()`to the interface and avoid referring to `ImmutableBTreePartition` directly.